src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -702,6 +746,24 @@ struct RemoveUnusedModuleElements : public Pass {
     module->removeTags([&](Tag* curr) {
       return !needed({ModuleElementKind::Tag, curr->name});
     });
+    module->removeMemories([&](Memory* curr) {
+      return !needed(ModuleElement(ModuleElementKind::Memory, curr->name));
+    });
+    module->removeTables([&](Table* curr) {
+      return !needed(ModuleElement(ModuleElementKind::Table, curr->name));
+    });
+    module->removeDataSegments([&](DataSegment* curr) {
+      return !needed(ModuleElement(ModuleElementKind::DataSegment, curr->name));
+    });
+    module->removeMemories([&](Memory* curr) {
+      return !needed({ModuleElementKind::Memory, curr->name});
+    });
+    module->removeTables([&](Table* curr) {
+      return !needed({ModuleElementKind::Table, curr->name});
+    });
+    module->removeDataSegments([&](DataSegment* curr) {
+      return !needed({ModuleElementKind::DataSegment, curr->name});
+    });
```

Comment:
Looks like the first and last 3 here are duplicates? The first three may have been intended to be deleted?